src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,167 @@
+package duke.parser;
+
+import java.io.IOException;
+
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidDescriptionException;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents the location where the commands entered into the program are
+ * processed and how should the <code>Ui</code>, or Seaward, should respond.
+ */
+public class Parser {
+
+    private static TaskList taskList;
```

Comment:
Should we extract out tasklist storage and UI? Could consider using Duke.java to control the logic flow instead of exposing storage, UI and tasklist all to parser?

Reply:
This is true since I am indeed exposing these classes to parser. I will try to address this issue when I have the time.